src/plugins/kibana/server/lib/schemas/resources/index_pattern_schema.js

```diff
@@ -6,6 +6,7 @@ module.exports = Joi.object({
   time_field_name: Joi.string(),
   interval_name: Joi.string(),
   not_expandable: Joi.boolean(),
+  fieldFilters: Joi.string().default('[]'),
```

Comment:
The code here (in the PR) defines a schema used by a new Kibana API. It's actually independent of the format that's ultimately stored in Elasticsearch.

The code you've linked to is older and interacts directly with Elasticsearch, which is why it matches exactly what you see in ES.

So the short answer is yes, `index_pattern_schema.js` should be snake_cased while `_index_pattern.js` should be camelCased.